src/intentLogic/getHeadlines.js

```diff
@@ -31,20 +31,33 @@ module.exports = function (isNewIntentFlag) {
         .then(asJson)
         .then((json) => {
             if (json.response.editorsPicks && json.response.editorsPicks.length >= attributes.moreOffset + PAGE_SIZE) {
-                var headlinesSpeech = randomMsg(speech.acknowledgement) + ((isNewIntent) ? speech.headlines.top : speech.headlines.more);
-
-                json.response.editorsPicks.slice(attributes.moreOffset, attributes.moreOffset+3).forEach(editorsPick => {
-                    headlinesSpeech += editorsPick.fields.headline + sound.transition;
-                    attributes.positionalContent.push(editorsPick.id);
-                });
-
-                headlinesSpeech += sound.break + speech.headlines.question;
-                this.emit(':ask', headlinesSpeech, speech.headlines.reprompt);
+				this.emit(':ask', generateHeadlinesSpeech(json), speech.headlines.reprompt);
             } else {
                 this.emit(':ask', speech.headlines.notfound);
             }
         })
         .catch(function (error) {
-            this.emit(':tell', headlinesSpeech, speech.headlines.notfound);
-        })
+            this.emit(':tell', speech.headlines.notfound);
+        });
+
+
+	var generateHeadlinesSpeech = (json) => {
+		const preamble = randomMsg(speech.acknowledgement) + ((isNewIntent) ? speech.headlines.top : speech.headlines.more);
+		const conclusion = sound.break + speech.headlines.reprompt;
+
+		var getHeadlines = () => {
+			return json.response.editorsPicks.slice(attributes.moreOffset, attributes.moreOffset+3).map(editorsPick =>
+				editorsPick.fields.headline + sound.transition
+			);
+		};
+
+		var setPositionalContent = () => {
+			json.response.editorsPicks.slice(attributes.moreOffset, attributes.moreOffset+3).forEach(editorsPick => {
+				attributes.positionalContent.push(editorsPick.id);
```

Comment:
Since we're always creating a new array for `positionalContent`, can we replace all this with:
```
attributes.positionalContent = 
  json.response.editorsPicks.slice(attributes.moreOffset, attributes.moreOffset+3).map(editorsPick => { return editorsPick.id } )
```